tests/topology.rs

```diff
@@ -38,7 +42,7 @@ fn into_message(event: Event) -> String {
         .to_string_lossy()
 }
 
-#[tokio::test(core_threads = 2)]
+#[tokio::test(threaded_scheduler)]
```

Comment:
Why change this one? It looked fine...

Reply:
I do not know why this was not worked with basic scheduler, I guess that because we have spawns in the test. Anyway, `threaded_scheduler` / `core_threads` is fine for now, but will be removed later. This PR only addresses `tokio::test` macro addition, otherwise will be hard review all changes.

Reply:
Ah, so it was broken, and `threaded_scheduler` instead of `core_threads = 2` fixes things? Just looks odd to me as this test already uses `#[tokio::test]` macro, and there are no changes to the internals of the test itself. I assumed it was fine (i.e. not broken), but if it is broken - makes sense.

I'd be surprised if our use of `spawn`s required multiple threads. That'd mean we have some major concurrency management flaws in the code.

Reply:
Yeah, a lot of tests had `core_threads = 2` before, I added this attribute when I moved them to `tokio::test` (because otherwise tests were hanged up). `threaded_scheduler` works because of `core_threads` with value more than 1 produce threaded scheduler in macro.